src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingRemotingMessageHandler.cs

```diff
@@ -0,0 +1,166 @@
+        /// <param name="messageHeaders">Request message headers.</param>
+        /// <param name="requestBody">Request message body</param>
+        /// <returns></returns>
+        public Task<byte[]> RequestResponseAsync(IServiceRemotingRequestContext requestContext, ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            return HandleAndTrackRequestAsync(messageHeaders, () => this.innerHandler.RequestResponseAsync(requestContext, messageHeaders, requestBody));
+        }
+
+        private void Initialize()
+        {
+            this.telemetryClient = new TelemetryClient();
+            this.baggageSerializer = new Lazy<DataContractSerializer>(() => new DataContractSerializer(typeof(IEnumerable<KeyValuePair<string, string>>)));
+
+            // TODO: SF should expose method name without the need to use reflection
+            this.methodMap = typeof(ServiceRemotingDispatcher)?.GetField("methodDispatcherMap", BindingFlags.Instance | BindingFlags.NonPublic)
+                                ?.GetValue(this.innerHandler) as IDictionary<int, ServiceMethodDispatcherBase>;
+        }
+
+        private async Task<byte[]> HandleAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, Func<Task<byte[]>> doHandleRequest)
+        {
+            // Create and prepare activity and RequestTelemetry objects to track this request.
+            RequestTelemetry rt = new RequestTelemetry();
+
+            if (messageHeaders.TryGetHeaderValue(ServiceRemotingLoggingStrings.ParentIdHeaderName, out string parentId) &&
+                messageHeaders.TryGetHeaderValue(ServiceRemotingLoggingStrings.RootIdHeaderName, out string rootId))
+            {
+                rt.Context.Operation.Id = rootId;
+                rt.Context.Operation.ParentId = parentId;
+            }
+
+            // Do our best effort in setting the request name. If we have the service context, add the service name. If
+            // we have the method map, add the method name to it.
+            if (this.serviceContext != null)
+            {
+                string methodName = null;
+                if (this.methodMap != null && this.methodMap.TryGetValue(messageHeaders.InterfaceId, out ServiceMethodDispatcherBase method))
+                {
+                    try
+                    {
+                        methodName = method.GetMethodName(messageHeaders.MethodId);
+                    }
+                    catch (KeyNotFoundException)
+                    {
+                    }
+                }
+
+                // Weird case, just use the numerical id as the method name
+                if (string.IsNullOrEmpty(methodName))
+                {
+                    methodName = messageHeaders.MethodId.ToString();
+                }
+
+                rt.Name = this.serviceContext.ServiceName + "/" + methodName;
+            }
+
+            Activity activity = Activity.Current;
```

Comment:
there is no activity before operation is started. So StartOperation should be done before adding baggage. However, then telemetry will be initialized BEFORE you add baggage to Activity. So you have to add baggage AND add properties to telemetry.Context.Properties manually